lib/stub_ui/lib/painting.dart

```diff
@@ -1629,10 +1634,15 @@ class Codec {
 /// unlikely that a factor that low will be sufficient to cache all decoded
 /// frames. The default value is `25.0`.
 ///
+/// If [targetWidth] or [targetHeight] are specified, image is resized to this. This is
```

Comment:
"image is resized to this" is a bit gramatically dubious, and "container" isn't clear here (what container?). How about:

```dart
  /// The [targetWidth] and [targetHeight] arguments specify the size of the output
  /// image, in image pixels. If they are not equal to the intrinsic dimensions of the
  /// image, then the image will be scaled after being decoded. If exactly one of
  /// these two arguments is set to [kDoNotResizeDimension], then the aspect ratio
  /// will be maintained while forcing the image to match the other given dimension.
  /// If both are [kDoNotResizeDimension], then the image maintains its real size.
```

I still think it would be more idiomatic to use `null` rather than `-1` in the API. It would avoid the need to introduce [kDoNotResizeDimension] to the public API. We would just pass `targetWidth ?? _kDoNotResizeDimension` in the implementation, so that the null doesn't make it to C++.

Reply:
I like this idea. Will use `null` here.